public/app/plugins/datasource/tempo/resultTransformer.test.ts

```diff
@@ -54,3 +54,70 @@ describe('transformFromOTLP()', () => {
     expect(res.data[0]).toMatchObject(otlpDataFrameFromResponse);
   });
 });
+
+describe('transformFromOTLP()', () => {
+  // Mock the console error so that running the test suite doesnt throw the error
+  const origError = console.error;
+  const consoleErrorMock = jest.fn();
+  afterEach(() => (console.error = origError));
+  beforeEach(() => (console.error = consoleErrorMock));
+
+  test('if passed bad data, will surface an error', () => {
+    const res = transformFromOTLP(
+      (badOTLPResponse.batches as unknown) as collectorTypes.opentelemetryProto.trace.v1.ResourceSpans[],
+      false
+    );
+
+    expect(res.data[0]).toBeFalsy();
+    expect(res?.error?.message).toContain('JSON is not valid OpenTelemetry format');
+    // if it does have resources, no error will be thrown
```

Comment:
Not sure what this next part is testing. I think it can be removed

Reply:
I think we should just remove it as it doesn't test any new behavior or new data. Given a falsy value (res.data[0]) and a truthy value (the resources object you pass in), of course it's going to be truthy